### PR TITLE
don't hardcode sockets per node to 2 in openmpi.py

### PR DESCRIPTION
**Pull request by boegel**
wrapper PR for https://github.com/hpcugent/vsc-mympirun/pull/127, which deals with merge conflict and failing test on Jenkins

### CODE REVIEW

**Comment by boegel** [comment]
all `mympirun` tests in `vsc-testing` still pass (on 10x3 cores on golett), so this should be good to go...

<details>

```
[11:52:07] vsc40023@node2471:~/vsc-testing/mympirun $ time ./run_all_tests.sh

*** 001_basic.sh ***

> mympirun hostname
>> with intel/2014b
>> with intel/2015a
>> with intel/2015b
>> with intel/2016a
>> with foss/2015a
>> with foss/2015b
>> with foss/2016a
>> with foss/2016b

>>> 001_basic.sh: PASS

*** 002_output.sh ***

> mympirun --output

>>> 002_output.sh: PASS

*** 003_options.sh ***

> with intel/2014b
>> mympirun --hybrid 1 hostname
>> mympirun --hybrid 2 hostname
>> mympirun --hybrid 4 hostname
>> mympirun --hybrid 16 hostname
>> mympirun --hybrid 32 hostname
>> mympirun --hybrid 64 hostname
>> mympirun --universe 1 hostname
>> mympirun --universe 2 hostname
>> mympirun --universe 4 hostname
>> mympirun --universe 16 hostname
>> mympirun --double hostname
>> mympirun --multi 2 hostname
>> mympirun --multi 3 hostname
>> mympirun --multi 4 hostname
> with intel/2015a
>> mympirun --hybrid 1 hostname
>> mympirun --hybrid 2 hostname
>> mympirun --hybrid 4 hostname
>> mympirun --hybrid 16 hostname
>> mympirun --hybrid 32 hostname
>> mympirun --hybrid 64 hostname
>> mympirun --universe 1 hostname
>> mympirun --universe 2 hostname
>> mympirun --universe 4 hostname
>> mympirun --universe 16 hostname
>> mympirun --double hostname
>> mympirun --multi 2 hostname
>> mympirun --multi 3 hostname
>> mympirun --multi 4 hostname
> with intel/2015b
>> mympirun --hybrid 1 hostname
>> mympirun --hybrid 2 hostname
>> mympirun --hybrid 4 hostname
>> mympirun --hybrid 16 hostname
>> mympirun --hybrid 32 hostname
>> mympirun --hybrid 64 hostname
>> mympirun --universe 1 hostname
>> mympirun --universe 2 hostname
>> mympirun --universe 4 hostname
>> mympirun --universe 16 hostname
>> mympirun --double hostname
>> mympirun --multi 2 hostname
>> mympirun --multi 3 hostname
>> mympirun --multi 4 hostname
> with intel/2016a
>> mympirun --hybrid 1 hostname
>> mympirun --hybrid 2 hostname
>> mympirun --hybrid 4 hostname
>> mympirun --hybrid 16 hostname
>> mympirun --hybrid 32 hostname
>> mympirun --hybrid 64 hostname
>> mympirun --universe 1 hostname
>> mympirun --universe 2 hostname
>> mympirun --universe 4 hostname
>> mympirun --universe 16 hostname
>> mympirun --double hostname
>> mympirun --multi 2 hostname
>> mympirun --multi 3 hostname
>> mympirun --multi 4 hostname
> with intel/2016b
>> mympirun --hybrid 1 hostname
>> mympirun --hybrid 2 hostname
>> mympirun --hybrid 4 hostname
>> mympirun --hybrid 16 hostname
>> mympirun --hybrid 32 hostname
>> mympirun --hybrid 64 hostname
>> mympirun --universe 1 hostname
>> mympirun --universe 2 hostname
>> mympirun --universe 4 hostname
>> mympirun --universe 16 hostname
>> mympirun --double hostname
>> mympirun --multi 2 hostname
>> mympirun --multi 3 hostname
>> mympirun --multi 4 hostname
> with intel/2017a
>> mympirun --hybrid 1 hostname
>> mympirun --hybrid 2 hostname
>> mympirun --hybrid 4 hostname
>> mympirun --hybrid 16 hostname
>> mympirun --hybrid 32 hostname
>> mympirun --hybrid 64 hostname
>> mympirun --universe 1 hostname
>> mympirun --universe 2 hostname
>> mympirun --universe 4 hostname
>> mympirun --universe 16 hostname
>> mympirun --double hostname
>> mympirun --multi 2 hostname
>> mympirun --multi 3 hostname
>> mympirun --multi 4 hostname
> with foss/2015a
>> mympirun --hybrid 1 hostname
>> mympirun --hybrid 2 hostname
>> mympirun --hybrid 4 hostname
>> mympirun --hybrid 16 hostname
>> mympirun --hybrid 32 hostname
>> mympirun --hybrid 64 hostname
>> mympirun --universe 1 hostname
>> mympirun --universe 2 hostname
>> mympirun --universe 4 hostname
>> mympirun --universe 16 hostname
>> mympirun --double hostname
>> mympirun --multi 2 hostname
>> mympirun --multi 3 hostname
>> mympirun --multi 4 hostname
> with foss/2015b
>> mympirun --hybrid 1 hostname
>> mympirun --hybrid 2 hostname
>> mympirun --hybrid 4 hostname
>> mympirun --hybrid 16 hostname
>> mympirun --hybrid 32 hostname
>> mympirun --hybrid 64 hostname
>> mympirun --universe 1 hostname
>> mympirun --universe 2 hostname
>> mympirun --universe 4 hostname
>> mympirun --universe 16 hostname
>> mympirun --double hostname
>> mympirun --multi 2 hostname
>> mympirun --multi 3 hostname
>> mympirun --multi 4 hostname
> with foss/2016a
>> mympirun --hybrid 1 hostname
>> mympirun --hybrid 2 hostname
>> mympirun --hybrid 4 hostname
>> mympirun --hybrid 16 hostname
>> mympirun --hybrid 32 hostname
>> mympirun --hybrid 64 hostname
>> mympirun --universe 1 hostname
>> mympirun --universe 2 hostname
>> mympirun --universe 4 hostname
>> mympirun --universe 16 hostname
>> mympirun --double hostname
>> mympirun --multi 2 hostname
>> mympirun --multi 3 hostname
>> mympirun --multi 4 hostname
> with foss/2016b
>> mympirun --hybrid 1 hostname
>> mympirun --hybrid 2 hostname
>> mympirun --hybrid 4 hostname
>> mympirun --hybrid 16 hostname
>> mympirun --hybrid 32 hostname
>> mympirun --hybrid 64 hostname
>> mympirun --universe 1 hostname
>> mympirun --universe 2 hostname
>> mympirun --universe 4 hostname
>> mympirun --universe 16 hostname
>> mympirun --double hostname
>> mympirun --multi 2 hostname
>> mympirun --multi 3 hostname
>> mympirun --multi 4 hostname

>>> 003_options.sh: PASS

*** 004_variables.sh ***

> export MYMPIRUN_BASEPATH='/'
> mympirun hostname

> export TEST_VARIABLE_ONE=one
> export TEST_VARIABLE_TWO=2
> mympirun env

> export MYMPIRUN_VARIABLESPREFIX=TEST_VARIABLE
> mympirun env

>>> 004_variables.sh: PASS

*** 005_sched.sh ***

> mympirun  hostname
> mympirun --sched=local hostname
> unset PBS_NODEFILE
> mympirun  hostname
> mympirun --sched=local hostname

>>> 005_sched.sh: PASS

*** 006_hello.sh ***

Hello world from processor node2471.golett.os, rank 0 out of 30 processors
Hello world from processor node2471.golett.os, rank 1 out of 30 processors
Hello world from processor node2471.golett.os, rank 2 out of 30 processors
Hello world from processor node2479.golett.os, rank 3 out of 30 processors
Hello world from processor node2479.golett.os, rank 4 out of 30 processors
Hello world from processor node2479.golett.os, rank 5 out of 30 processors
Hello world from processor node2483.golett.os, rank 6 out of 30 processors
Hello world from processor node2483.golett.os, rank 7 out of 30 processors
Hello world from processor node2483.golett.os, rank 8 out of 30 processors
Hello world from processor node2489.golett.os, rank 9 out of 30 processors
Hello world from processor node2489.golett.os, rank 10 out of 30 processors
Hello world from processor node2489.golett.os, rank 11 out of 30 processors
Hello world from processor node2496.golett.os, rank 12 out of 30 processors
Hello world from processor node2496.golett.os, rank 13 out of 30 processors
Hello world from processor node2496.golett.os, rank 14 out of 30 processors
Hello world from processor node2499.golett.os, rank 15 out of 30 processors
Hello world from processor node2499.golett.os, rank 16 out of 30 processors
Hello world from processor node2499.golett.os, rank 17 out of 30 processors
Hello world from processor node2499.golett.os, rank 18 out of 30 processors
Hello world from processor node2499.golett.os, rank 19 out of 30 processors
Hello world from processor node2499.golett.os, rank 20 out of 30 processors
Hello world from processor node2499.golett.os, rank 21 out of 30 processors
Hello world from processor node2499.golett.os, rank 22 out of 30 processors
Hello world from processor node2499.golett.os, rank 23 out of 30 processors
Hello world from processor node2499.golett.os, rank 24 out of 30 processors
Hello world from processor node2499.golett.os, rank 25 out of 30 processors
Hello world from processor node2499.golett.os, rank 26 out of 30 processors
Hello world from processor node2499.golett.os, rank 27 out of 30 processors
Hello world from processor node2499.golett.os, rank 28 out of 30 processors
Hello world from processor node2499.golett.os, rank 29 out of 30 processors

>>> 006_hello.sh: PASS

TEST RESULT: all 6 passed!

real	5m26.908s
user	1m24.489s
sys	0m43.263s
```

</details>

**Comment by JensTimmerman** [comment]
`pinning_override` is never used?

File: lib/vsc/mympirun/mpi/openmpi.py [hunk 1]
```diff
@@ -67,7 +95,7 @@ def pinning_override(self):
         self.log.debug("pinning_override: type %s ", override_type)
 
         ranktxt = ""
-        sockets_per_node = 2
+        sockets_per_node = self.determine_sockets_per_node()
```
Comment (JensTimmerman) [inline lib/vsc/mympirun/mpi/openmpi.py]: instead of sockets per node, shouldn't you be using the sockets available in the cpuset for the process?

Comment (boegel) [inline lib/vsc/mympirun/mpi/openmpi.py]: In order to do the pinning as requested, `mympirun` needs to know the number of sockets in the node (currently 2 in all our systems, but it shouldn't be hardcoded)

This is only required when using OpenMPI and when `--overridepin` is used, see https://github.com/hpcugent/vsc-mympirun/blob/develop/lib/vsc/mympirun/mpi/mpi.py#L324 and https://github.com/hpcugent/vsc-mympirun/blob/develop/lib/vsc/mympirun/mpi/mpi.py#L1034

**Pull request merged**